lib/src/Uncertainty/Bayesian/GaussianNonLinearCalibration.cxx

```diff
@@ -389,7 +389,15 @@ Point GaussianNonLinearCalibration::run(const Sample & inputObservations,
   LeastSquaresProblem problem(residualFunction);
   algorithm_.setVerbose(true);
   algorithm_.setProblem(problem);
-  algorithm_.setStartingPoint(getCandidate());
+  try
+  {
+    // If the solver is single start, we can use its setStartingPoint method
+    algorithm_.setStartingPoint(getCandidate());
+  }
+  catch (NotYetImplementedException &) // setStartingPoint is not implemented in MultiStart
+  {
+    LOGWARN(OSS() << "Candidate=" << getCandidate() << " is ignored because algorithm_ is MultiStart.");
```

Comment:
Update the message: "... because the algorithm has no setStartingPoint method" or something similar (no explicit mention of multistart, but you can insert algorithm_.getImplementation()->getClassName() in the message)